evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -772,43 +786,57 @@ where
 
         // If the depth is 0, then this is the root call terminating
         if data.journaled_state.depth() == 0 {
-            for (address, expecteds) in &self.expected_calls {
-                for (expected, actual_count) in expecteds {
-                    let ExpectedCallData { calldata, gas, min_gas, value, count } = expected;
+            for (address, calldatas) in &self.expected_calls {
+                for (calldata, (expected, actual_count)) in calldatas {
+                    let ExpectedCallData { gas, min_gas, value, count, call_type } = expected;
                     let calldata = calldata.clone();
-                    let expected_values = [
-                        Some(format!("data {calldata}")),
-                        value.map(|v| format!("value {v}")),
-                        gas.map(|g| format!("gas {g}")),
-                        min_gas.map(|g| format!("minimum gas {g}")),
-                    ]
-                    .into_iter()
-                    .flatten()
-                    .join(" and ");
-                    if count.is_none() {
-                        if *actual_count == 0 {
-                            return (
-                                InstructionResult::Revert,
-                                remaining_gas,
-                                format!("Expected at least one call to {address:?} with {expected_values}, but got none")
+                    match call_type {
```

Comment:
Match depending of mode:
- `Count`: `count` Must be **strictly** `actual_count`, if not we should fail.
- `NonCount`: `actual_count` must be **at least equal to** `count` (therefore, if we see that `count` is bigger than `actual_count`, we should fail as we didn't see enough calls).

For reference:
- `count`: Amount of times a call must be seen.
- `actual_count` Amount of times a call was actually seen.